authorize.go

```diff
@@ -121,32 +121,45 @@ func (s *Server) HandleAuthorizeRequest(w *Response, r *http.Request) *Authorize
 		HttpRequest: r,
 	}
 
-	// must have a valid client
-	ret.Client, err = w.Storage.GetClient(r.Form.Get("client_id"))
-	if err != nil {
-		w.SetErrorState(E_SERVER_ERROR, "unable to get client", ret.State)
-		w.InternalError = err
-		return nil
-	}
-	if ret.Client == nil {
-		w.SetErrorState(E_UNAUTHORIZED_CLIENT, "client not found", ret.State)
-		return nil
-	}
-	if ret.Client.GetRedirectURI() == "" {
-		w.SetErrorState(E_UNAUTHORIZED_CLIENT, "client redirect URI not found", ret.State)
-		return nil
+	clientIDs := r.Form["client_id"]
+	comboClient := &ComboClient{
+		Audience: clientIDs,
+		Clients:  make([]Client, 0, len(clientIDs)),
 	}
+	for _, id := range clientIDs {
+		// must have a valid client
+		var cl Client
+		cl, err = w.Storage.GetClient(id)
+		if err != nil {
+			w.SetErrorState(E_SERVER_ERROR, "unable to get client", ret.State)
+			w.InternalError = err
+			return nil
+		}
+		if cl == nil {
+			w.SetErrorState(E_UNAUTHORIZED_CLIENT, "client not found", ret.State)
+			return nil
+		}
+		if cl.GetRedirectURI() == "" {
+			w.SetErrorState(E_UNAUTHORIZED_CLIENT, "client redirect URI not found", ret.State)
+			return nil
+		}
+		comboClient.Clients = append(comboClient.Clients, cl)
 
-	// check redirect uri, if there are multiple client redirect uri's
-	// don't set the uri
-	if ret.RedirectUri == "" && FirstUri(ret.Client.GetRedirectURI(), s.Config.RedirectUriSeparator) == ret.Client.GetRedirectURI() {
-		ret.RedirectUri = FirstUri(ret.Client.GetRedirectURI(), s.Config.RedirectUriSeparator)
+		// check redirect uri, if there are multiple client redirect uri's
+		// don't set the uri
+		if ret.RedirectUri == "" && FirstUri(cl.GetRedirectURI(), s.Config.RedirectUriSeparator) == cl.GetRedirectURI() {
+			ret.RedirectUri = FirstUri(cl.GetRedirectURI(), s.Config.RedirectUriSeparator)
```

Comment:
This will only return the last client's redirect URI then? It'll be overwritten every loop